modules/n1ql/pages/n1ql-language-reference/typefun.adoc

```diff
@@ -1,36 +1,225 @@
+----
+
+[source,json]
+----
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
+== ISATOM(expression)
+
+=== Description
 
 Returns True if expression is a Boolean, number, or string, otherwise returns MISSING, NULL or false.
 
-ISBOOLEAN(expression)
+=== Examples
+====
+[source,n1ql]
+----
+SELECT ISATOM(true) AS `boolean`,
+       ISATOM(MISSING) AS `missing`,
+       ISATOM(NULL) AS `null`,
+       ISATOM(123) AS `number`,
+       ISATOM("hello world") AS `string`,
+       ISATOM([1, 2, 3]) AS `array`,
+       ISATOM({"hello": "world"}) AS `object`;
+----
+
+[source,json]
+----
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
+== ISBOOLEAN(expression)
+
+=== Description
 
 Returns True if expression is a Boolean, otherwise returns MISSING, NULL or false.
 
-ISNUMBER(expression)
+=== Examples
+====
+[source,n1ql]
+----
+SELECT ISBOOLEAN(true) AS `boolean`,
+       ISBOOLEAN(MISSING) AS `missing`,
+       ISBOOLEAN(NULL) AS `null`,
+       ISBOOLEAN(123) AS `number`,
+       ISBOOLEAN("hello world") AS `string`,
+       ISBOOLEAN([1, 2, 3]) AS `array`,
+       ISBOOLEAN({"hello": "world"}) AS `object`;
+----
+
+[source,json]
+----
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
+== ISNUMBER(expression)
+
+=== Description
 
 Returns True if expression is a number, otherwise returns MISSING, NULL or false.
 
-ISOBJECT(expression)
+=== Examples
+====
+[source,n1ql]
+----
+SELECT ISNUMBER(true) AS `boolean`,
+       ISNUMBER(MISSING) AS `missing`,
+       ISNUMBER(NULL) AS `null`,
+       ISNUMBER(123) AS `number`,
+       ISNUMBER("hello world") AS `string`,
+       ISNUMBER([1, 2, 3]) AS `array`,
+       ISNUMBER({"hello": "world"}) AS `object`;
+----
+
+[source,json]
+----
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
+== ISOBJECT(expression)
+
+=== Description
 
 Returns True if expression is an object, otherwise returns MISSING, NULL or false.
 
+=== Examples
+====
+[source,n1ql]
+----
+SELECT ISOBJECT(true) AS `boolean`,
+       ISOBJECT(MISSING) AS `missing`,
+       ISOBJECT(NULL) AS `null`,
+       ISOBJECT(123) AS `number`,
+       ISOBJECT("hello world") AS `string`,
+       ISOBJECT([1, 2, 3]) AS `array`,
+       ISOBJECT({"hello": "world"}) AS `object`;
+----
+
+[source,json]
+----
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
 ISSTRING(expression)
 
 Returns True if expression is a string, otherwise returns MISSING, NULL or false.
 
+== ISSTRING(expression)
+
+=== Description
```

Comment:
Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks if the supplied expression is a string.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```